src/district0x/server/utils.cljs

```diff
@@ -59,4 +59,9 @@
 (defn print-take! [ch]
   (take! ch println))
 
-(def tx-sent? (comp d0x-shared-utils/sha3? second))
\ No newline at end of file
+(def tx-sent? (comp d0x-shared-utils/sha3? second))
+
+(defn tx-failed? [tx]
+  (when-let [ftx (first tx)]
```

Comment:
You can just check if first is not nil. First on tuple is always reserved only for error